lib/elixir/test/elixir/kernel/expansion_test.exs

```diff
@@ -2861,6 +2878,29 @@ defmodule Kernel.ExpansionTest do
     Macro.prewalk(expr, &Macro.update_meta(&1, cleaner))
   end
 
+  @bitstring_modifiers [
+    :integer,
+    :float,
+    :binary,
+    :utf8,
+    :utf16,
+    :utf32,
+    :native,
+    :signed,
+    :bitstring,
+    :little
+  ]
+
+  defp clean_bit_modifiers(expr) do
+    Macro.prewalk(expr, fn
+      {expr, meta, atom} when expr in @bitstring_modifiers and is_atom(atom) ->
+        {expr, meta, nil}
```

Comment:
This part isn't very satisifying, do you have an idea how we could handle this elegantly?
Else we'd get errors like

```elixir
     left:  {:<<>>, [], [{:"::", [], [{:x, [], []}, {:-, [], [{:utf32, [], nil}, {:little, [], nil}]}]}]}
     right: {:<<>>, [], [{:"::", [], [{:x, [], []}, {:-, [], [{:utf32, [], nil}, {:little, [], Kernel.ExpansionTest}]}]}]}
```